### DISCUSSION TITLE
How does wazuh detect the cpe of a product and check against vulnerabilities.

### DISCUSSION THREAD

**Discussion by mai1x9** [wazuh/wazuh#26772] [Q&A]
Hello, I am curious, how wazuh will detect vulnerabilities for some of the software. Does it use CPE string, if so then how will it extract cpe of the product.

For Eg: If I consider, Vendor: `Palo Alto Networks` product, `GlobalProtect`, the cpe string  looks like `cpe:2.3:a:paloaltonetworks:globalprotect: .............`, but how does wazuh knows the CPE string.

When we query the Registry Information for installed applications from windows, all it gives is Display Name: `GlobalProtect` and Publisher is `Palto Alto Networks`,  So I am curious how wazuh actually queries database.

**Comment by MiguelazoDS** [comment]
Hi @mai1x9, 

For Windows and Mac where the NVD vulnerability feed is used, we use the CPE to match against a package. The sanitization process occurs in another stage where the CPEs are processed, then that information is uploaded to CTI and is ready to use by the scanner module. 

Is your question related to fact that the vulnerabilities for that packages are not detected? That's role of translations, the syscollector module detects the packages through the Windows registry and receives the information you mentioned.

**Reply by mai1x9** [reply]
@MiguelazoDS Thanks for the reply.

Let me be clear. I was asking, if I install Palo Alto global protect, then windows saves the application with Display Name as `GlobalProtect` and publisher is `Palto Alto Networks`.

But the cpe string is: `cpe:2.3:a:paloaltonetworks:globalprotect:...`. So my question is how does wazuh know it is `cpe:2.3:a:paloaltonetworks:globalprotect:..` or how does wazuh predict it.?

> That's role of translations, the syscollector module detects the packages through the Windows registry and receives the information you mentioned.

Is this manaully need to defined so wazuah understands the cpe string is `cpe:2.3:a:paloaltonetworks:globalprotect:..`  or it applies some prediction algorithm etc.

Also I believe with just `GlobalProtect` and publisher is `Palto Alto Networks`, we cannot directly search NVD database. If we know cpe string, i believe we get better results. So my question is how does wazuh actually predict or make that cpe string before searching NVD feed.

**Reply by MiguelazoDS** [reply]
I'll asume you get `Palo Alto Networks` not `Palto...` and  yes, you're right, with that package name and vendor the package can not be scanned. The translation is already tracked in a issue an it'll addressed pretty soon. 

> So my question is how does wazuh actually predict or make that cpe string before searching NVD feed.

Once the translation is done, it will be uploaded to the content and automatically (depending on you feed update interval setting) it will be updated with that information. 
It's not a prediction, it's a manual task, for some cases we just convert the package name and vendor to lowercase and that may work for a reduced number of packages, but most of them need a translation. 

I'll talk to the team to let them know about this.

**Reply by mai1x9** [reply]
@MiguelazoDS yup sounds great got it. So its a translation that actually helps in converting the registry pulled information about software into a proper CPE string before checking with NVD. 

Looking forward to hear from your team.

-----------

Another follow up to the question, Let's say I have a chrome version `95.XX.XX.XX.`, on checking NVD database, its showing 840+ vulnerabilities.

Does wazuh shows the same number of vulnerabilities or it applies some algorithm to remove false positives or group vulnerabilities etc. If it requires a separate discussion I will create a new one.

**Comment by MiguelazoDS** [comment]
I've just notified the team in charge to perform the translation about this issue. 

About the other question, sometimes the NVD reports, as you mention, some false negatives or positives, and for that case there's also a manual work over the content in CTI to sanitize that information. So, sometimes the number will match with the NVD, and sometime it will not. 

For that reason we're also working on this https://github.com/wazuh/wazuh/issues/26098 studying this other vulnerability feed to improve the accuracy of our reports.

**Reply by MiguelazoDS** [reply]
Of course, there may be inconsistencies on our side too, there are so many feed and each of them have different criteria to report their vulnerabilities and during the normalization something may be wrongly parsed or interpreted, so if you find that something doesn't make any sense please let us know and we'll study the case.

**Reply by mai1x9** [reply]
I understand. Thanks for clarifications. 

> I've just notified the team in charge to perform the translation about this issue.

I have read about cpe-helper on wazuh, is it the one responsible for translation. If so, then I have no doubts. I have got full understanding. I was just wondering if translation is different from cpe-helper ?

**Reply by MiguelazoDS** [reply]
That was the old fashion to do translations, but it has several issues, one pretty critical, the scanner was not prepared to accept modifications in the cpe-helper https://github.com/wazuh/wazuh/issues/10631, we had to manually modify the databases to let the scanner think, the translation content was updated. 

Also that didn't escalate, each change in the cpe-helper works only locally for the one who did the change. Now once we found (or someone) notice a translation is missing, we added to the CTI content and then the new content is available for everyone. 

So answering the question, yes, it's different but it is the same purpose.

**Reply by mai1x9** [reply]
Thanks, understood. Can you please point out any references for the translations or issues so i can have better insights on translations.

**Reply by MiguelazoDS** [reply]
@mai1x9 do you have the information from Syscollector for that package (and any other you also find necessary) so I can create a tracking issue? 

Basically, as the translations are managed internally by the team in charge, we create an issue on this repository that will show the progress to the community.

**Comment by sebasfalcone** [comment]
@mai1x9 If possible provide us with the [OS](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_os_info), [package](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_packages_info) and [hotfixes](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_hotfix_info) information as Migue requested

Inactive thread, please re-open it if needed
